samples/Java/Basic/ReceiveUsingOffset/src/main/java/com/microsoft/azure/eventhubs/samples/receiveusingoffset/ReceiveUsingOffset.java

```diff
@@ -7,35 +7,37 @@
 import com.microsoft.azure.eventhubs.ConnectionStringBuilder;
 import com.microsoft.azure.eventhubs.EventData;
 import com.microsoft.azure.eventhubs.EventHubClient;
+import com.microsoft.azure.eventhubs.EventHubException;
+import com.microsoft.azure.eventhubs.EventPosition;
 import com.microsoft.azure.eventhubs.EventHubRuntimeInformation;
 import com.microsoft.azure.eventhubs.PartitionReceiver;
-import com.microsoft.azure.eventhubs.EventHubException;
 
 import java.io.IOException;
 import java.nio.charset.Charset;
 import java.util.concurrent.ExecutionException;
-import java.util.function.BiConsumer;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 
 public class ReceiveUsingOffset {
     public static void main(String[] args)
             throws EventHubException, ExecutionException, InterruptedException, IOException {
 
-        final String namespaceName = "----ServiceBusNamespaceName-----";
-        final String eventHubName = "----EventHubName-----";
-        final String sasKeyName = "-----SharedAccessSignatureKeyName-----";
-        final String sasKey = "---SharedAccessSignatureKey----";
-        final ConnectionStringBuilder connStr = new ConnectionStringBuilder(namespaceName, eventHubName, sasKeyName, sasKey);
+        final ConnectionStringBuilder connStr = new ConnectionStringBuilder()
+                .setNamespaceName("----ServiceBusNamespaceName-----") // to target National clouds - use .setEndpoint(URI)
+                .setEventHubName("----EventHubName-----")
+                .setSasKeyName("-----SharedAccessSignatureKeyName-----")
+                .setSasKey("---SharedAccessSignatureKey----");
 
-        final EventHubClient ehClient = EventHubClient.createFromConnectionStringSync(connStr.toString());
+        final ExecutorService executorService = Executors.newSingleThreadExecutor();
+        final EventHubClient ehClient = EventHubClient.createSync(connStr.toString(), executorService);
 
         final EventHubRuntimeInformation eventHubInfo = ehClient.getRuntimeInformation().get();
         final String partitionId = eventHubInfo.getPartitionIds()[0]; // get first partition's id
         
         final PartitionReceiver receiver = ehClient.createEpochReceiverSync(
                 EventHubClient.DEFAULT_CONSUMER_GROUP_NAME,
                 partitionId,
-                PartitionReceiver.START_OF_STREAM,
-                false,
+                EventPosition.fromStartOfStream(), // Specifying OFFSET - is the most performant way to create Receivers.
```

Comment:
It might be unclear that `EventPosition.fromStartOfStream()` is setting an offset.